host/sgx/sgxload.c

```diff
@@ -402,9 +439,19 @@ oe_result_t oe_sgx_create_enclave(
                 enclave_error);
 
         secs->base = (uint64_t)base;
+
+        if (context->create_zero_base_enclave)
+        {
+            /* Returned base has to be same as requested start_address */
+            if ((uint64_t)base != (uint64_t)base_address)
+                OE_RAISE_MSG(
+                    OE_PLATFORM_ERROR,
+                    "enclave_create failed at requested start address");
+            secs->base = (uint64_t)OE_ADDRESS_ZERO;
+        }
     }
-#endif // OEHOSTMR
-    *enclave_addr = base ? (uint64_t)base : secs->base;
```

Comment:
This logic is specific to `OEHOSTMR`. Can you double check?

Reply:
Reverted to previous logic.